airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/CatalogHelpers.java

```diff
@@ -322,16 +325,23 @@ public static Set<StreamTransform> getCatalogDiff(final AirbyteCatalog oldCatalo
         .forEach(descriptor -> {
           final AirbyteStream streamOld = descriptorToStreamOld.get(descriptor);
           final AirbyteStream streamNew = descriptorToStreamNew.get(descriptor);
+
+          final ConfiguredAirbyteStream stream = configuredCatalog.getStreams().stream()
+              .filter(s -> Objects.equals(s.getStream().getNamespace(), descriptor.getNamespace())
+                  && s.getStream().getName().equals(descriptor.getName()))
+              .findFirst().get();
```

Comment:
[`.get()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Optional.html#get()) will throw a `NoSuchElementException` if called on an empty `Optional`.

What should happen here if we have nothing?  Should `streamTransoforms.add` still be called? Is nothing an error case?

Reply:
I don't think this is a case that should happen, but if it does, I think we should default to calling the transform non-breaking. I'll make that change.